### PR TITLE
 Temporary fix for problem with length limit of 5000 in MLflow

### PR DESCRIPTION
**Pull request by PhilipMay**
PR for #1340

### CODE REVIEW

File: optuna/integration/mlflow.py [hunk 1]
```diff
@@ -121,4 +121,16 @@ def __call__(self, study: optuna.study.Study, trial: optuna.trial.FrozenTrial) -
...
+            # This is a temporary fix on Optuna side. It avoids an error with user
+            # attributes that are too long. It should be fixed on MLflow side later.
+            # When it is fixed on MLflow side this codeblock can be removed.
+            # see https://github.com/optuna/optuna/issues/1340
+            # see https://github.com/mlflow/mlflow/issues/2931
+            max_mlflow_tag_length = 5000
+            for key, value in tags.items():
+                value = str(value)  # make sure it is a string
+                if len(value) > max_mlflow_tag_length:
+                    tags[key] = value[:max_mlflow_tag_length]
```
Comment (hvy) [inline optuna/integration/mlflow.py:134]: `textwrap.shorten` with `placeholder` could be used but it's a detail, https://docs.python.org/3/library/textwrap.html#textwrap.shorten.

Comment (PhilipMay) [reply optuna/integration/mlflow.py:134]: I will add that tonight.

Comment (PhilipMay) [reply optuna/integration/mlflow.py]: I made the change.

**Comment by PhilipMay** [comment]
IMO this is ready to be merged.

**Pull request merged**